lib/mongoid/extensions/raw_value.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Wrapper class used when a value cannot be casted in evolve method.
+module Mongoid
+  def RawValue(*args)
+    RawValue.new(*args)
+  end
+
+  class RawValue
+
+    attr_reader :raw_value
+
+    def initialize(raw_value)
+      @raw_value = raw_value
+    end
+
+    # Delegate all missing methods to the raw value.
+    #
+    # @param [ String, Symbol ] method_name The name of the method.
+    # @param [ Array ] args The arguments passed to the method.
+    ruby2_keywords def method_missing(method_name, *args, &block)
```

Comment:
This implementation is not necessary. The topic of proxies is a complex one, for example proxies generally need to derive from BasicObject. Please remove `method_missing` and `respond_to_missing?` from here and if you have a use case which requires RawValue to be a proxy please file a ticket describing it.

What might be useful is a `to_s` or `inspect` override which would call the raw value's `to_s` or `inspect` and prefix the output with `RawValue:`.

Reply:
Done.